lib/info.dart

```diff
@@ -529,17 +539,17 @@ InfoKind kindFromString(String kind) {
 }
 
 /// A simple visitor for information produced by the dart2js compiler.
-abstract class InfoVisitor<T> {
-  T visitAll(AllInfo info);
-  T visitProgram(ProgramInfo info);
-  T visitLibrary(LibraryInfo info);
-  T visitClass(ClassInfo info);
-  T visitField(FieldInfo info);
-  T visitConstant(ConstantInfo info);
-  T visitFunction(FunctionInfo info);
-  T visitTypedef(TypedefInfo info);
-  T visitClosure(ClosureInfo info);
-  T visitOutput(OutputUnitInfo info);
+abstract class InfoVisitor<T, C> {
```

Comment:
I wonder whether it is worth adding the context argument though. I fear this add a tax everywhere, but it is used pretty rarely. In many cases it is simple to track the context as an internal a state within the visitor, could we do that here instead?

IIUC, in the code below, that might only involve a change to add a `_context` field, and change a line in `_diffList` from:

```dart
       oldNames[oldName].accept(this, newNames[oldName]);
```

to something like:
```dart
       this._context = newNames[oldName];
       oldNames[oldName].accept(this);
```

WDYT?